kotlinx-coroutines-test/common/test/RunTestTest.kt

```diff
@@ -408,4 +408,67 @@ class RunTestTest {
     fun testCoroutineCompletingWithoutDispatch() = runTest(timeout = Duration.INFINITE) {
         launch(Dispatchers.Default) { delay(100) }
     }
+
+    /**
+     * Tests that [runTest] cleans up the exception handler even if it threw on initialization.
+     *
+     * This test must be run manually, because it writes garbage to the log.
+     */
+    @Test
+    @Ignore
+    fun testExceptionCaptorCleanedUpOnPreliminaryExit(): TestResult = testResultChain({
```

Comment:
If you see fit, you can write a JVM-only test that invokes `Thread.setDefaultUncaughtExceptionHandler` to avoid stderr cluttering

Reply:
Good idea, thanks!